src/Essentials/src/WebAuthenticator/WebAuthenticator.shared.cs

```diff
@@ -3,19 +3,62 @@
 using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading.Tasks;
+using System.ComponentModel;
+using Microsoft.Maui.Essentials;
+using Microsoft.Maui.Essentials.Implementations;
 
 namespace Microsoft.Maui.Essentials
 {
+	public interface IWebAuthenticator
+	{
+		Task<WebAuthenticatorResult> AuthenticateAsync(Uri url, Uri callbackUrl);
+
+		Task<WebAuthenticatorResult> AuthenticateAsync(WebAuthenticatorOptions webAuthenticatorOptions);
+
+#if IOS || MACCATALYST
```

Comment:
@mattleibow I'd like another set of eyes on this.  It's perhaps a bit odd but I feel that we need to be able to still invoke the static methods to call through the current implementation's instance rather than rely on our implementations having singleton state properties here or there.

We can chat tuesday about this too as I think this has implications to other refactoring PR's

Reply:
I think this will fail at runtime if I, for example, just have an implementation for `net6.0` or `netstandard`. Not useful, but might be for example a `net6.0-windows`, `net6.0-android` and `net6.0`. Then running on iOS will be a crash with an invalid IL.

So maybe we need a `interface IPlatformWebAuthenticator : IWebAuthenticator` that people can optionally implement.

Then the 

```cs
internal static bool OpenUrl(Uri uri) => (Current as IPlatformWebAuthenticator ).OpenUrlCallback(uri);
```